### PR TITLE
Update Elasticsearch views documentation.

### PR DESCRIPTION
**Pull request by wwajerowicz**
screencapture-file-Users-wajerowi-code-nexus-docs-target-site-docs-delta-api-current-views-elasticsearch-view-api-html-2021-05-12-15_57_45

### CODE REVIEW

File: docs/src/main/paradox/docs/delta/api/current/assets/views/elasticsearch/deprecated.json [hunk 1]
```diff
@@ -0,0 +1,73 @@
+{
+  "@context": [
```
Comment (bogdanromanx) [inline docs/src/main/paradox/docs/delta/api/current/assets/views/elasticsearch/deprecated.json:2]: this file doesn't seem right...

File: docs/src/main/paradox/docs/delta/api/current/assets/views/elasticsearch/create-put.sh [hunk 2]
```diff
@@ -0,0 +1,33 @@
+curl -XPUT \
+-H "Content-Type: application/json" \
+"http://localhost:8080/v1/views/myorg/myproj/myview" -d \
+'{
+    "@type": [
+        "ElasticSearchView"
+    ],
+    "mapping": {
```
Comment (bogdanromanx) [inline docs/src/main/paradox/docs/delta/api/current/assets/views/elasticsearch/create-put.sh:8]: could you please correct the indenting here?

File: docs/src/main/paradox/docs/delta/api/current/assets/views/elasticsearch/create.sh [hunk 3]
```diff
@@ -0,0 +1,33 @@
+curl -XPOST \
+-H "Content-Type: application/json" \
+"http://localhost:8080/v1/views/myorg/myproj" -d \
+'{
+    "@type": [
+        "ElasticSearchView"
+    ],
+    "mapping": {
+    "dynamic": false,
```
Comment (bogdanromanx) [inline docs/src/main/paradox/docs/delta/api/current/assets/views/elasticsearch/create.sh:9]: same as above, there seems to be an indenting issue here.

File: docs/src/main/paradox/docs/delta/api/current/assets/views/elasticsearch/update.sh [hunk 4]
```diff
@@ -0,0 +1,33 @@
+curl -XPUT \
+-H "Content-Type: application/json" \
+"http://localhost:8080/v1/views/myorg/myproj/myview?rev=1" -d \
+'{
+    "@type": [
+        "ElasticSearchView"
+    ],
+    "mapping": {
+    "dynamic": false,
```
Comment (bogdanromanx) [inline docs/src/main/paradox/docs/delta/api/current/assets/views/elasticsearch/update.sh:9]: indenting issue

File: docs/src/main/paradox/docs/delta/api/current/views/elasticsearch-view-api.md [hunk 5]
```diff
@@ -2,17 +2,19 @@
...
-An asynchronous process gets trigger for every view. This process can be visualized as a pipeline with different stages. 
+An asynchronous process gets trigger for every view. This process can be visualized as a pipeline with different stages.
 
 The first stage is the input of the pipeline: a stream of events scoped for the project where the view was created.
 
-The last stage takes the JSON document, generated through the pipeline steps, and stores it as a Document in an ElasticSearch index
+The last stage takes the JSON document, generated through the pipeline steps, and stores it as a Document in an
+ElasticSearch index
 
 [![ElasticSearchView pipeline](../assets/views/elasticsearch_pipeline.png "ElasticSearchView pipeline")](../assets/views/elasticsearch_pipeline.png)
```
Comment (bogdanromanx) [inline docs/src/main/paradox/docs/delta/api/current/views/elasticsearch-view-api.md]: the image you posted in the PR description does not show this picture... maybe the path is wrong?

File: docs/src/main/paradox/docs/delta/api/current/views/elasticsearch-view-api.md [hunk 6]
```diff
@@ -2,17 +2,19 @@
...
-The documents created on each view are isolated from documents created on other views by using different ElasticSearch indices.
+The documents created on each view are isolated from documents created on other views by using different ElasticSearch
+indices.
 
 A default view gets automatically created when the project is created but other views can be created.
 
 ## Processing pipeline
 
-An asynchronous process gets trigger for every view. This process can be visualized as a pipeline with different stages. 
+An asynchronous process gets trigger for every view. This process can be visualized as a pipeline with different stages.
```
Comment (bogdanromanx) [inline docs/src/main/paradox/docs/delta/api/current/views/elasticsearch-view-api.md:12]: gets triggered

File: docs/src/main/paradox/docs/delta/api/current/views/elasticsearch-view-api.md [hunk 7]
```diff
@@ -2,17 +2,19 @@
...
 
 The first stage is the input of the pipeline: a stream of events scoped for the project where the view was created.
 
-The last stage takes the JSON document, generated through the pipeline steps, and stores it as a Document in an ElasticSearch index
+The last stage takes the JSON document, generated through the pipeline steps, and stores it as a Document in an
```
Comment (bogdanromanx) [inline docs/src/main/paradox/docs/delta/api/current/views/elasticsearch-view-api.md:16]: in the corresponding ElasticSearch index.

File: docs/src/main/paradox/docs/delta/api/current/views/elasticsearch-view-api.md [hunk 8]
```diff
@@ -28,30 +30,39 @@ The last stage takes the JSON document, generated through the pipeline steps, an
...
-  escaped string value of the key `_original_source`. If false, the resource's payload will be stored normally in the ElasticSearch document. The default value is `false`.
-- `{includeMetadata}`: Boolean - If true, the resource's nexus metadata (`_constrainedBy`, `_deprecated`, ...) will be 
+- `_elasticsearch mapping_`: Json object - Defines the value types for the Json keys, as stated at the
+  @link:[ElasticSearch mapping documentation](https://www.elastic.co/guide/en/elasticsearch/reference/current/indices-put-mapping.html#indices-put-mapping){
+  open=new }.
+- `_elasticssearch settings_`: Json object(Optional) - defines Elasticsearch
+  @link:[index settings](https://www.elastic.co/guide/en/elasticsearch/reference/current/indices-create-index.html#create-index-settings){
+  open=new } for the underlying Elasticsearch index. Default settings are applied, if not specified.
+- `{sourceAsText}`: Boolean - If true, the resource's payload will be stored in the ElasticSearch document as a single
+  escaped string value of the key `_original_source`. If false, the resource's payload will be stored normally in the
```
Comment (bogdanromanx) [inline docs/src/main/paradox/docs/delta/api/current/views/elasticsearch-view-api.md:52]: under the key

File: docs/src/main/paradox/docs/delta/api/current/views/elasticsearch-view-api.md [hunk 9]
```diff
@@ -94,30 +105,220 @@ include the resource metadata fields.
...
+Payload
+:   @@snip [payload.json](../assets/views/elasticsearch/payload.json)
+
+Response
+:   @@snip [created.json](../assets/views/elasticsearch/created.json)
+
+### Create using PUT
+
+This alternative endpoint to create a view is useful in case the json payload does not contain an `@id` but you want to
+specify one. The @id will be specified in the last segment of the endpoint URI.
```
Comment (bogdanromanx) [inline docs/src/main/paradox/docs/delta/api/current/views/elasticsearch-view-api.md:135]: `@id`

File: docs/src/main/paradox/docs/delta/api/current/views/elasticsearch-view-api.md [hunk 10]
```diff
@@ -94,30 +105,220 @@ include the resource metadata fields.
...
+
+```
+PUT /v1/views/{org_label}/{project_label}/{schema_id}/{view_id}
+  {...}
+```
+
+Note that if the payload contains an @id different from the `{view_id}`, the request will fail.
```
Comment (bogdanromanx) [inline docs/src/main/paradox/docs/delta/api/current/views/elasticsearch-view-api.md:142]: `@id`

File: docs/src/main/paradox/docs/delta/api/current/views/elasticsearch-view-api.md [hunk 11]
```diff
@@ -94,30 +105,220 @@ include the resource metadata fields.
...
+
+### Update
+
+This operation overrides the payload.
+
+In order to ensure a client does not perform any changes to a resource without having had seen the previous revision of
+the view, the last revision needs to be passed as a query parameter.
+
+```
+PUT /v1/views/{org_label}/{project_label}/{schema_id}/{view_id}?rev={previous_rev}
```
Comment (bogdanromanx) [inline docs/src/main/paradox/docs/delta/api/current/views/elasticsearch-view-api.md:163]: the endpoint is wrong

File: docs/src/main/paradox/docs/delta/api/current/views/elasticsearch-view-api.md [hunk 12]
```diff
@@ -94,30 +105,220 @@ include the resource metadata fields.
...
+Response
+:   @@snip [created.json](../assets/views/elasticsearch/created.json)
+
+### Create using PUT
+
+This alternative endpoint to create a view is useful in case the json payload does not contain an `@id` but you want to
+specify one. The @id will be specified in the last segment of the endpoint URI.
+
+```
+PUT /v1/views/{org_label}/{project_label}/{schema_id}/{view_id}
```
Comment (bogdanromanx) [inline docs/src/main/paradox/docs/delta/api/current/views/elasticsearch-view-api.md:138]: the endpoint is wrong

File: docs/src/main/paradox/docs/delta/api/current/views/elasticsearch-view-api.md [hunk 13]
```diff
@@ -94,30 +105,220 @@ include the resource metadata fields.
...
+:   @@snip [updated.json](../assets/views/elasticsearch/updated.json)
+
+## Tag
+
+Links a view revision to a specific name.
+
+Tagging a view is considered to be an update as well.
+
+```
+POST /v1/views/{org_label}/{project_label}/{schema_id}/{view_id}/tags?rev={previous_rev}
```
Comment (bogdanromanx) [inline docs/src/main/paradox/docs/delta/api/current/views/elasticsearch-view-api.md:194]: the endpoint is wrong

File: docs/src/main/paradox/docs/delta/api/current/views/elasticsearch-view-api.md [hunk 14]
```diff
@@ -94,30 +105,220 @@ include the resource metadata fields.
...
+... where
+
+- `{previous_rev}`: is the last known revision number for the resource.
+- `{name}`: String - label given to the view at specific revision.
+- `{rev}`: Number - the revision to link the provided `{name}`.
+
+**Example**
+
+Request
+:   @@snip [tag.sh](../assets/views/elasticsearch/tag.sh)
```
Comment (bogdanromanx) [inline docs/src/main/paradox/docs/delta/api/current/views/elasticsearch-view-api.md:210]: this file is empty according to this PR

File: docs/src/main/paradox/docs/delta/api/current/views/elasticsearch-view-api.md [hunk 15]
```diff
@@ -94,30 +105,220 @@ include the resource metadata fields.
...
+:   @@snip [search-payload.json](../assets/views/elasticsearch/search-payload.json)
 
 Response
-:   @@snip [elastic-view-search.json](../assets/views/elastic-view-search.json)
+:   @@snip [search-results.json](../assets/views/elasticsearch/search-results.json)
+
+## Fetch tags
+
+```
+GET /v1/views/{org_label}/{project_label}/{schema_id}/{view_id}/tags?rev={rev}&tag={tag}
```
Comment (bogdanromanx) [inline docs/src/main/paradox/docs/delta/api/current/views/elasticsearch-view-api.md:305]: endpoint is wrong

**Review by bogdanromanx — CHANGES_REQUESTED** [review]

File: docs/src/main/paradox/docs/delta/api/current/views/elasticsearch-view-api.md [hunk 16]
```diff
@@ -28,30 +30,39 @@ The last stage takes the JSON document, generated through the pipeline steps, an
   "sourceAsText": {sourceAsText},
   "includeMetadata": {includeMetadata},
   "includeDeprecated": {includeDeprecated},
-  "mapping": _elasticsearch mapping_
+  "mapping": _elasticsearch mapping_,
+  "settings": _elasticsearch settings_,
+  "permission": "views/query"
```
Comment (umbreak) [inline docs/src/main/paradox/docs/delta/api/current/views/elasticsearch-view-api.md:35]: `"permission": "{permission}"`

File: docs/src/main/paradox/docs/delta/api/current/views/elasticsearch-view-api.md [hunk 17]
```diff
@@ -28,30 +30,39 @@ The last stage takes the JSON document, generated through the pipeline steps, an
...
+  @link:[index settings](https://www.elastic.co/guide/en/elasticsearch/reference/current/indices-create-index.html#create-index-settings){
+  open=new } for the underlying Elasticsearch index. Default settings are applied, if not specified.
+- `{sourceAsText}`: Boolean - If true, the resource's payload will be stored in the ElasticSearch document as a single
+  escaped string value of the key `_original_source`. If false, the resource's payload will be stored normally in the
+  ElasticSearch document. The default value is `false`.
+- `{includeMetadata}`: Boolean - If true, the resource's nexus metadata (`_constrainedBy`, `_deprecated`, ...) will be
   stored in the ElasticSearch document. Otherwise it won't. The default value is `false`.
 - `{includeDeprecated}`: Boolean - If true, deprecated resources are also indexed. The default value is `false`.
 - `{someid}`: Iri - The @id value for this view.
+- `permission`: String(Optional) - permission required to query this view. Defaults to `views/query`.
```
Comment (umbreak) [inline docs/src/main/paradox/docs/delta/api/current/views/elasticsearch-view-api.md:58]: `{permission}`

File: docs/src/main/paradox/docs/delta/api/current/assets/views/elasticsearch/create-put.sh [hunk 18]
```diff
@@ -0,0 +1,33 @@
+curl -XPUT \
+-H "Content-Type: application/json" \
+"http://localhost:8080/v1/views/myorg/myproj/myview" -d \
+'{
+    "@type": [
```
Comment (umbreak) [inline docs/src/main/paradox/docs/delta/api/current/assets/views/elasticsearch/create-put.sh:5]: No need for an array, `"@type": "ElasticSearchView"` would be more readable (same on create-post.sh)

File: docs/src/main/paradox/docs/delta/api/current/assets/views/elasticsearch/create.sh [hunk 19]
```diff
@@ -0,0 +1,33 @@
+curl -XPOST \
+-H "Content-Type: application/json" \
+"http://localhost:8080/v1/views/myorg/myproj" -d \
+'{
+    "@type": [
```
Comment (umbreak) [inline docs/src/main/paradox/docs/delta/api/current/assets/views/elasticsearch/create.sh:5]: same

**Review by umbreak — CHANGES_REQUESTED** [review]

**Review by umbreak — COMMENTED** [review]
Why there is a title called `Endpoints` here? It is not consistent with the rest of the documentation. Should be dropped

File: docs/src/main/paradox/docs/delta/api/current/views/elasticsearch-view-api.md (hunk 14 above)
Comment (umbreak) [reply docs/src/main/paradox/docs/delta/api/current/views/elasticsearch-view-api.md:210]: And what the picture shows is a wrong curl request too

**Review by umbreak — CHANGES_REQUESTED** [review]

File: docs/src/main/paradox/docs/delta/api/current/assets/views/elasticsearch/deprecate.sh [hunk 20]
```diff
@@ -0,0 +1 @@
+curl -DELETE "http://localhost:8080/v1/views/myorg/myproj/myview?rev=5"
```
Comment (umbreak) [inline docs/src/main/paradox/docs/delta/api/current/assets/views/elasticsearch/deprecate.sh:1]: wrong. It should be:

```
curl -X DELETE
   "http://localhost:8080/v1/views/myorg/myproj/myview?rev=5
```

**Review by umbreak — CHANGES_REQUESTED** [review]

**Review by umbreak — COMMENTED** [review]
A `Fetch offset` endpoint should be added

**Comment by umbreak** [comment]
On the image, there is no tabs for the `Fetch tags` section. Is that an image mistake only?

**Comment by umbreak** [comment]
We should add another ```@@@note``` like the one we have on the `update` section for the deprecate section:
```
Deprecating a view deletes the view index, making the view not searchable.
```

File: docs/src/main/paradox/docs/delta/api/current/views/elasticsearch-view-api.md (hunk 17 above)
Comment (umbreak) [inline docs/src/main/paradox/docs/delta/api/current/views/elasticsearch-view-api.md:58]: The fact that you write `String(Optional)` here seems correct to me, but it is not consistent with the rest of the documentation. Either we change it everywhere else, or you align this one

**Review by umbreak — CHANGES_REQUESTED** [review]

File: docs/src/main/paradox/docs/delta/api/current/views/elasticsearch-view-api.md [hunk 21]
```diff
@@ -90,34 +100,226 @@ include the resource metadata fields.
 }
 ```
 
-## Endpoints
+### Create using POST
```
Comment (imsdu) [inline docs/src/main/paradox/docs/delta/api/current/views/elasticsearch-view-api.md:103]: the different options should be `##`

**Review by imsdu — CHANGES_REQUESTED** [review]

File: docs/src/main/paradox/docs/delta/api/current/views/elasticsearch-view-api.md [hunk 22]
```diff
@@ -56,7 +56,7 @@ where...
 - `{someid}`: Iri - The @id value for this view.
 - `{permission}`: String(Optional) - permission required to query this view. Defaults to `views/query`.
 
-### Example
+## Example
```
Comment (imsdu) [inline docs/src/main/paradox/docs/delta/api/current/views/elasticsearch-view-api.md:59]: `**Example**`

**Review by imsdu — CHANGES_REQUESTED** [review]

**Review by imsdu — APPROVED** [review]

File: docs/src/main/paradox/docs/delta/api/current/views/elasticsearch-view-api.md [hunk 23]
```diff
@@ -49,13 +48,13 @@ where...
...
 - `{sourceAsText}`: Boolean - If true, the resource's payload will be stored in the ElasticSearch document as a single
-  escaped string value of the key `_original_source`. If false, the resource's payload will be stored normally in the
+  escaped string value under the key `_original_source`. If false, the resource's payload will be stored normally in the
   ElasticSearch document. The default value is `false`.
 - `{includeMetadata}`: Boolean - If true, the resource's nexus metadata (`_constrainedBy`, `_deprecated`, ...) will be
   stored in the ElasticSearch document. Otherwise it won't. The default value is `false`.
 - `{includeDeprecated}`: Boolean - If true, deprecated resources are also indexed. The default value is `false`.
 - `{someid}`: Iri - The @id value for this view.
-- `permission`: String(Optional) - permission required to query this view. Defaults to `views/query`.
+- `{permission}`: String(Optional) - permission required to query this view. Defaults to `views/query`.
```
Comment (umbreak) [inline docs/src/main/paradox/docs/delta/api/current/views/elasticsearch-view-api.md]: there is this `String(Optional)`...

**Review by umbreak — CHANGES_REQUESTED** [review]
The titles for the operations are wrong. They are h3 (`####`) while they should be h2 (`####`). You have removed the `Endpoints` title, but not updated the rest

**Review by umbreak — CHANGES_REQUESTED** [review]
The titles for the operations are wrong. They are h3 (`####`) while they should be h2 (`####`). You have removed the `Endpoints` title, but not updated the rest

**Review by umbreak — APPROVED** [review]

**Review by bogdanromanx — APPROVED** [review]

**Pull request merged**